ffi/Makefile

```diff
@@ -1,9 +1,25 @@
+UNAME_OS := $(shell uname -s)
+ifeq ($(UNAME_OS),Darwin)
+	OS_FLAG=darwin
+endif
+ifeq ($(UNAME_OS),Linux)
+	OS_FLAG=linux
+endif
+
+UNAME_ARCH := $(shell uname -s)
+ifeq ($(UNAME_ARCH),i386)
+	ARCH_FLAG=i386
```

Comment:
arch flag should be 386 and Not i386

Reply:
also to include windows - see here for inspiration

ifeq ($(OS),Windows_NT)
    OS_FLAG=windows
    ifeq ($(PROCESSOR_ARCHITEW6432),AMD64)
        ARCH_FLAG=amd64
    else
        ifeq ($(PROCESSOR_ARCHITECTURE),AMD64)
            ARCH_FLAG=amd64
        endif
        ifeq ($(PROCESSOR_ARCHITECTURE),x86)
            ARCH_FLAG=386
        endif
    endif
else
    //// the above
endif

see https://stackoverflow.com/questions/714100/os-detecting-makefile